bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/display/controls/BahmniObservationsController.java

```diff
@@ -109,18 +125,45 @@ public BahmniObservationsController(BahmniObsService bahmniObsService, ConceptSe
         } else {
             observations = bahmniObsService.getObservationsForPatientProgram(patientProgramUuid, rootConceptNames, obsIgnoreList);
         }
+        makeComplexDataNull(observations, loadComplexData);
         sendObsToGroovyScript(rootConceptNames, observations);
         return observations;
     }
 
     @RequestMapping(method = RequestMethod.GET, params = {"observationUuid"})
     @ResponseBody
     public BahmniObservation get(@RequestParam(value = "observationUuid") String observationUuid,
-                                 @RequestParam(value = "revision", required = false) String revision) {
+                                 @RequestParam(value = "revision", required = false) String revision,
+                                 @RequestParam(value = "loadComplexData", required = false) Boolean loadComplexData) {
+        BahmniObservation observation;
         if (ObjectUtils.equals(revision, LATEST)) {
-            return bahmniObsService.getRevisedBahmniObservationByUuid(observationUuid);
+            observation = bahmniObsService.getRevisedBahmniObservationByUuid(observationUuid);
+        } else {
+            observation = bahmniObsService.getBahmniObservationByUuid(observationUuid);
         }
-        return bahmniObsService.getBahmniObservationByUuid(observationUuid);
+        makeComplexDataNull(observation, loadComplexData);
+        return observation;
+    }
+
+    private void makeComplexDataNull(Collection<BahmniObservation> observations, Boolean loadComplexData) {
+        if (!isNull(loadComplexData) && loadComplexData == false) {
+            observations.forEach(observation -> {
+                if (!isNull(observation.getGroupMembers()) && observation.getGroupMembers().size() > 0) {
```

Comment:
Duplicate code can be removed by calling `makeComplexDataNull(BahmniObservation observation, Boolean loadComplexData)`

Reply:
Refactored the code.